HeterogeneousCore/CUDATest/plugins/TestCUDAProducerGPUKernel.cu

```diff
@@ -97,7 +97,7 @@ cudautils::device::unique_ptr<float[]> TestCUDAProducerGPUKernel::runAlgo(const
 
   auto d_c = cudautils::make_device_unique<float[]>(NUM_VALUES, stream);
   auto current_device = cudautils::currentDevice();
-  edm::LogVerbatim("TestHeterogeneousEDProducerGPU")
+  cudautils::LogVerbatim("TestHeterogeneousEDProducerGPU")
       << "  " << label << " GPU launching kernels device " << current_device << " CUDA stream " << stream;
```

Comment:
Given that the message is emitted in the destructor, how about adding scopes around the messages to get them emitted at the expected time? I.e.
```cpp
{
  cudautils::LogVerbatim(...) ...
}
```

Reply:
Since the variable is anonymous, I think the destructor is called immediately, not at the end of the scope.

Reply:
See https://stackoverflow.com/a/2298796/2050986 .

Reply:
Right, of course.